architecture/Safe Control.md

```diff
@@ -0,0 +1,74 @@
+
+- Finally, the user is redirected to the action/motion page where they will finalize the process by following a link that we provide for them to execute their transactions on the other network via a Live Monitor app that keeps track of cross-chain Safe transactions. Meanwhile, the action/motion page will show a tag indicating that the transaction needs an action from the user. Once the transaction is executed by the user in the live monitor app, the tag will be updated to show that the transaction was executed.
+
+- To summarize the process, it would be something like this: Saga > Colony contracts (`makeArbitraryTransactions`) > Home Bridge (`requireToPassMessage`) (Gnosis network) > Foreign Bridge (Ethereum or Binance network) > Transaction is manually executed on the foreign chain by the user via the Live Monitor app > Zodiac Bridge Module (`executeTransaction`) > Selected safe executes the transactions.
+
+### Block ingestor
+
+- We look for `ColonyMetadata` events in the case of adding or removing a safe and then we parse them in client-side in order to know whether the event is emitted due to a change of linked safes, colony name, address book, etc.
+
+- For safe transactions, we catch the `ArbitraryTransaction` event and parse it to know whether the transaction was created by the user or not and if it's already present in the DB. In the first case, the transaction might not be "created" by the user if the arbitrary transaction was emitted by the motion finalizing. In the second case, the transaction might already be present in the DB if the user sent multiple transactions (Every transaction included by the user in the dialog will emit its own event).
+
+## Live Monitor App
+
+For Ethereum: https://alm-gnosis-eth.colony.io/
+For Binance: https://alm-gnosis-bsc.colony.io/
+
+These services are hosted and maintained by us. It was previously developed and maintained by [TokenBridge](https://docs.tokenbridge.net/) but they seem to ahve abandoned it, and with this app being essential for the users to execute their transactions, we've decided it to maintain it ourselves.
+
+## Testing Safe Control locally
+
+Testing this feature on the dev env is possible due to the docker container called `safe`, which will be used to deploy a foreign chain, home and foreign bridges, and a safe with tokens and 1 NFT. The addresses to each of the relevant contracts will be printed in the console once the container is up and running.
+
+Once they are printed out, you must include them in the `.env` file as the `.env.example` indicates.
+
+With that, you can start testing the feature as you would in QA or production with the only difference being that you don't need to execute the transactions in the live monitor app since the transactions will be executed automatically by the mock contracts, therefore, locally the transactions will always be shown as `Completed` from the get-go.
```

Comment:
I would add here the point that the safes you enter via the Safe control ui aren't the ones that are interacted with. If I recall correctly, isn't it also the case that we don't verify that the transaction is actually executed by the safe? 

I remember from testing that transferring tokens from the safe to myself did not increase the token balance in my wallet (but I may be forgetting/misremembering some of the details).

Reply:
Good idea 👍 I'm gonna add it in